### PR TITLE
Sw dspdc 900 create orchestration workflow

### PR DESCRIPTION
**Pull request by snwessel**
Setting up an orchestration workflow for Encode ([Jira Ticket](https://broadinstitute.atlassian.net/browse/DSPDC-900)).

* Related to this PR in monster-deploy: https://github.com/broadinstitute/monster-deploy/pull/60

### CODE REVIEW

File: orchestration/values.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  storageClass: standard
+  downloadSize: 1Gi
+  extractSize: 10Gi
+cron:
+  enable: true
+  # Every Sunday at 3:30am
+  schedule: '30 3 * * Sun'
+  timezone: 'America/New_York'
+dataflow:
+  region: us-west-1
```
Comment (snwessel) [inline orchestration/values.yaml:11]: I updated the region, but the rest of these values were copied from https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/values.yaml

File: orchestration/values.schema.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        "useFlexRS": { "type": "boolean" },
+        "pipelineVersion": { "type": "string" }
+      },
+      "required": ["project", "region", "tmpBucketName",
+        "subnetName", "workerAccount", "workerMachineType",
+        "autoscaling", "useFlexRS", "pipelineVersion"]
+      }
+    },
+  "required": ["gcs", "serviceAccount", "volumes", "cron", "dataflow"]
+}
```
Comment (snwessel) [inline orchestration/values.schema.json:70]: Copied from https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/values.schema.json

File: orchestration/templates/run-transformation-pipeline.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          {{- with .autoscaling }}
+          - --autoscalingAlgorithm=THROUGHPUT_BASED
+          - --numWorkers={{ .minWorkers }}
+          - --maxNumWorkers={{ .maxWorkers }}
+          {{- end }}
+          {{- if .useFlexRS }}
+          - --flexRSGoal=COST_OPTIMIZED
+          {{- else }}
+          - --experiments=shuffle_mode=service
+          {{- end }}
```
Comment (snwessel) [inline orchestration/templates/run-transformation-pipeline.yaml:38]: The args here are copied from the clinvar transformation: https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/templates/run-dataflow.yaml

File: orchestration/templates/run-extraction-pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          {{- with .autoscaling }}
+          - --autoscalingAlgorithm=THROUGHPUT_BASED
+          - --numWorkers={{ .minWorkers }}
+          - --maxNumWorkers={{ .maxWorkers }}
+          {{- end }}
+          {{- if .useFlexRS }}
+          - --flexRSGoal=COST_OPTIMIZED
+          {{- else }}
+          - --experiments=shuffle_mode=service
+          {{- end }}
```
Comment (snwessel) [inline orchestration/templates/run-extraction-pipeline.yaml:37]: The args here are copied from the clinvar transformation: https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/templates/run-dataflow.yaml

File: orchestration/templates/cron-workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      - name: main
+        steps:
+          - - name: run-main-template
+              templateRef:
+                name: ingest-encode-data
+                template: main
+              arguments:
+                parameters:
+                  - name: gcs-prefix
+                    value: '{{ template "argo.timestamp" }}'
```
Comment (snwessel) [inline orchestration/templates/cron-workflow.yaml:26]: I based this off of https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/templates/cron-workflow.yaml

File: orchestration/templates/run-extraction-pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          - name: gcs-bucket
+          - name: output-prefix
+      container:
+        image: us.gcr.io/broad-dsp-gcr-public/encode-extraction:{{ $version }}
+        command: []
+        args:
+          - --runner=dataflow
+          {{- $bucket := "{{inputs.parameters.gcs-bucket}}" }}
+             - --inputPrefix=gs://{{ $bucket }}/{{ "{{inputs.parameters.input-prefix}}" }}
+          - --outputPrefix=gs://{{ $bucket }}/{{ "{{inputs.parameters.output-prefix}}" }}
```
Comment (danxmoran) [inline orchestration/templates/run-extraction-pipeline.yaml:21]: Double-check that this is the actual arg expected by the pipeline. I remember that the two are inconsistent; one uses "prefix" and the other uses "path", I think.

File: orchestration/values.schema.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    "serviceAccount": {
+      "type": "object",
+      "properties": {
+        "k8sName": { "type": "string" },
+        "googleName": { "type": "string" }
+      },
+      "required": ["k8sName", "googleName"]
+    },
+    "volumes": {
```
Comment (danxmoran) [inline orchestration/values.schema.json:20]: Don't need this `volumes` sub-schema

File: orchestration/values.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+volumes:
+  storageClass: standard
+  downloadSize: 1Gi
+  extractSize: 10Gi
```
Comment (danxmoran) [inline orchestration/values.yaml:4]: Don't need the default values for `volumes` either

File: orchestration/values.yaml (hunk 1 above)
Comment (danxmoran) [reply orchestration/values.yaml:11]: We can go more lightweight for ENCODE, I think. Let's start with `n1-standard-1` and see if we need to bump things up.

File: orchestration/templates/cron-workflow.yaml (hunk 5 above)
Comment (danxmoran) [reply orchestration/templates/cron-workflow.yaml:26]: Ah, wait: `argo.timestamp` won't be defined in this chart. The easiest thing to do would be to copy-paste the `_helpers.tpl` from clinvar. It'll eventually get bundled into a common chart that both can depend on...

File: orchestration/templates/_helpers.tpl [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+{{/* Common retry settings to apply to flaky steps. */}}
+{{- define "argo.retry" }}
+retryStrategy:
+  retryPolicy: Always
+  backoff:
+    duration: 1s
+    factor: 2
+    maxDuration: 5m
+{{- end -}}
```
Comment (danxmoran) [inline orchestration/templates/_helpers.tpl:27]: As you suggested in https://github.com/DataBiosphere/clinvar-ingest/pull/37#pullrequestreview-397554289, feel free to delete this file in favor of depending on the new common-templates chart.

**Pull request merged**